src/lib.rs

```diff
@@ -513,6 +561,21 @@ impl<T: Asset> AssetAnimator<T> {
         }
     }
 
+    /// Ticks the tween, if present. See [`Tweenable::tick`] for details.
+    pub fn tick(
+        &mut self,
+        delta: Duration,
+        target: &mut T,
+        entity: Entity,
+        event_writer: &mut EventWriter<TweenCompleted>,
```

Comment:
This forces a change on the `EventWriter` (when the caller derefs the `ResMut`) even when there's no animation. This should probably take a `&mut ResMut` instead. Same for `Animator<T>::tick()`.

Reply:
Hmmm, I don't understand. Where are ResMuts involved? This is an event writer that's being passed in by reference which means it shouldn't be getting derefed.

Reply:
Hum sorry my bad I misread indeed, I thought about the general issue and didn't check that `EventWriter` is passed by value in the system. Nevermind, please ignore that.

Reply:
No worries lol :)